src/pulsar/mod.rs

```diff
@@ -0,0 +1,238 @@
+use std::{borrow::Cow, collections::BTreeMap};
+
+use testcontainers::{
+    core::{CmdWaitFor, ContainerPort, ContainerState, ExecCommand, Mount, WaitFor},
+    Image, TestcontainersError,
+};
+
+const NAME: &str = "apachepulsar/pulsar";
+const TAG: &str = "2.10.6";
+
+const PULSAR_PORT: ContainerPort = ContainerPort::Tcp(6650);
+const ADMIN_PORT: ContainerPort = ContainerPort::Tcp(8080);
+
+/// Module to work with [`Apache Pulsar`] inside of tests.
+/// **Requires protoc to be installed, otherwise will not build.**
+///
+/// This module is based on the official [`Apache Pulsar docker image`].
+///
+/// # Example
+/// ```
+/// use testcontainers_modules::{pulsar, testcontainers::runners::SyncRunner};
+///
+/// let pulsar = pulsar::Pulsar::default().start().unwrap();
+/// let http_port = pulsar.get_host_port_ipv4(6650).unwrap();
+///
+/// // do something with the running pulsar instance..
+/// ```
+///
+/// [`Apache Pulsar`]: https://github.com/apache/pulsar
+/// [`Apache Pulsar docker image`]: https://hub.docker.com/r/apachepulsar/pulsar/
+#[derive(Debug, Clone)]
+pub struct Pulsar {
+    data_mount: Mount,
+    env: BTreeMap<String, String>,
+    admin_commands: Vec<Vec<String>>,
+}
+
+impl Default for Pulsar {
+    /**
+     * Starts an in-memory instance in dev mode, with horrible token values.
+     * Obviously not to be emulated in production.
+     */
+    fn default() -> Self {
+        Self {
+            data_mount: Mount::tmpfs_mount("/pulsar/data"),
+            env: BTreeMap::new(),
+            admin_commands: vec![],
+        }
+    }
+}
+
+impl Pulsar {
+    /// Add configuration parameter to Pulsar `conf/standalone.conf`
+    pub fn with_config(mut self, name: impl Into<String>, value: impl Into<String>) -> Self {
+        self.env
+            .insert(format!("PULSAR_PREFIX_{}", name.into()), value.into());
+        self
+    }
```

Comment:
The documentation might be a bit confusing, because it's implemented utilizing ENV variables, while use can utilize mounting or `CopyTo` functionality for configuration file

Perhaps it should be `with_config_env` method? mentioning that it's an env variable with predefined prefix in accordance with https://pulsar.apache.org/docs/next/getting-started-docker/

WDYT?

Reply:
Good point! And also that clarifies that configuration is done with py script, which sometimes behaves differently that writing config directly into file